packages/data-transport-layer/src/services/l1-ingestion/handlers/state-batch-appended.ts

```diff
@@ -67,6 +68,18 @@ export const handleEventsStateBatchAppended: EventHandlerSet<
     }
   },
   storeEvent: async (entry, db) => {
+    // Defend against situations where we missed an event because the RPC provider
+    // (infura/alchemy/whatever) is missing an event.
+    if (entry.stateRootBatchEntry.index > 0) {
+      if (
```

Comment:
5 line `if` condition with a nested await makes it hard to read. You need to unroll it in your head. Doing things like this makes sense in solidity to save gas but we should optimize for readability in JS

Reply:
Fixed